### PR TITLE
Added Press Back Button Again to Exit App Functionality

### LINKED ISSUE #730: Adding Press Back Button Again To Exit App Functionality
**Describe the issue**
Currently, the user exits the app as soon as he presses the back button while on Navigation Activity. Since the app requires the user to press the back button to navigate up, the user can often, by mistake, exit the app. To prevent this from happening we should add a toast message that asks the user to press the back button again to confirm that the user surely wants to exit the app.

**To Reproduce or Feature details**
1. Press the back button while on Navigation Activity.
2. Notice that we exit the app without confirmation.

**Current behavior**

App exits without toast message with a single press of the back button

https://user-images.githubusercontent.com/17739006/105395662-fe5d7000-5c44-11eb-95ca-2735cb5e15dc.mp4

**Expected behavior**

Toast message shows up asking the user to press the back button again. If the user presses the button again in a certain time limit, only then the app exit.

https://user-images.githubusercontent.com/17739006/105395320-9b6bd900-5c44-11eb-9b44-87f5cef04ea0.mp4

**Smartphone (please complete as much of the following information as possible):**
 - Should run on all supported Android devices.

### PR DESCRIPTION
**Pull request by Prakhar-Agarwal-byte**
@cpg 
I have added the feature as requested in the issue. Now, first, a toast pops up asking the user to press the back button again, and if the user presses the back button again in 2 seconds, only then the app exit. Fixes #730.

### CODE REVIEW

File: src/main/res/values/strings.xml [hunk 1]
```diff
@@ -344,5 +344,6 @@
     <string name="size_largest">Size (largest first)</string>
     <string name="size_smallest">Size (smallest first)</string>
     <string name="file_type">File type</string>
+    <string name="message_toast_press_back_again">Press back again to exit</string>
```
Comment (chirag-jn) [inline src/main/res/values/strings.xml:348]: Please add translations in other languages.

**Review by chirag-jn — CHANGES_REQUESTED** [review]

**Comment by chirag-jn** [comment]
Made some changes in the code @Prakhar-Agarwal-byte. Please pull and complete the requested changes.

File: src/main/res/values/strings.xml (hunk 1 above)
Comment (prakharagarwal-dev) [reply src/main/res/values/strings.xml:348]: Done

Comment (Prakhar-Agarwal-byte) [reply src/main/res/values/strings.xml:347]: Not something related to this PR, but for a few days, I am not able to login to the Amahi app. It's showing a connection error. Earlier it was working fine.

**Comment by Prakhar-Agarwal-byte** [comment]
@chirag-jn
Not something related to this PR, but for a few days, I am not able to login to the Amahi app. It's showing a connection error. Earlier it was working fine. Is there some problem with the server?

**Comment by chirag-jn** [comment]
> @chirag-jn
> Not something related to this PR, but for a few days, I am not able to login to the Amahi app. It's showing a connection error. Earlier it was working fine. Is there some problem with the server?

Welcome to Amahi server works fine for me. Try reinstalling the app or creating new credentials on the Amahi website? @Prakhar-Agarwal-byte

**Comment by Prakhar-Agarwal-byte** [comment]
> > @chirag-jn
> > Not something related to this PR, but for a few days, I am not able to login to the Amahi app. It's showing a connection error. Earlier it was working fine. Is there some problem with the server?
> 
> Welcome to Amahi server works fine for me. Try reinstalling the app or creating new credentials on the Amahi website? @Prakhar-Agarwal-byte

I guess there was some problem with authentication. I signed out and logged in again on the website and it started worked again. Thanks!

**Comment by Prakhar-Agarwal-byte** [comment]
> > @chirag-jn
> > Not something related to this PR, but for a few days, I am not able to login to the Amahi app. It's showing a connection error. Earlier it was working fine. Is there some problem with the server?
> 
> Welcome to Amahi server works fine for me. Try reinstalling the app or creating new credentials on the Amahi website? @Prakhar-Agarwal-byte

I guess there was some problem with authentication. I signed out and logged in again on the website and it started worked again. Thanks!

**Comment by cpg** [comment]
We have had some issues with certificate renewals that caused connection errors. It all should be working well now.

File: src/main/res/values/strings.xml (hunk 1 above)
Comment (prakharagarwal-dev) [reply src/main/res/values/strings.xml:348]: Please approve changes

Comment (prakharagarwal-dev) [reply src/main/res/values/strings.xml:348]: @chirag-jn 
I have done the required changes already. Please approve the PR.

**Pull request merged**